Microsoft/WebTemplateStudio
# client calls extension instead of CoreTS to get frameworks

Task:https://microsoftgarage.visualstudio.com/Intern%20GitHub/_sprints/taskboard/WTS%20Team/Intern%20GitHub/Y19-VS/07

(this PR did not address extension calls CoreTS CLI, instead of call CoreTS API)
Ideally, client should call extension via message passing, rather than call CoreTS api directly. There are few API calls to CoreTS in our client right now. This PR is to address the **getFrameworks** API call to CoreTS.  Thus, in my PR: 
- from client, it sends getFrameworks command to extension
- in clinet, removed thunk action (since we don't need to resolve the promise from an api call then disptach update the store action)
- in extension, I added a module to listen and return response to getFrameworks command
- in client App.tsx, it listens to response(frameworks data) from extension and update the store with the response

**How to test:**
1. start the extension (once with preview mode, and the other time without preview mode)
2. the select frameworks page should still be the same
3. the extensions should still be able to generate templates

![frameworks](https://user-images.githubusercontent.com/12874317/60684993-8d354c80-9e55-11e9-8489-d45fe8c3c4f8.PNG)
